src/librustc_incremental/calculate_svh/hasher.rs

```diff
@@ -50,23 +73,28 @@ impl Hasher for IchHasher {
     }
 
     #[inline]
+    fn write_u8(&mut self, i: u8) {
+        self.write_uleb128(i as u64);
```

Comment:
For a hash like this to be correct, it has to be unambiguous. My personal benchmark for this, the one I find most intuitive, is: If we were serializing this data instead of hashing it, would we be able to correctly deserialize it again? In this case yes, because we always know whether we are reading a `u8` or a `u16` or something else next. A problem only arises if we have a bunch of, say, `u16`s in a row but they are encoded with an ambiguous variable length encoding.